traefik.go

```diff
@@ -114,13 +137,19 @@ Complete documentation is available at https://traefik.io`,
 
 	traefikConfiguration.ConfigFile = toml.ConfigFileUsed()
 
-	kv, err := CreateKvSource(traefikConfiguration)
+	kv, err = CreateKvSource(traefikConfiguration)
 	if err != nil {
 		fmtlog.Println(err)
 		os.Exit(-1)
 	}
 
-	if kv != nil {
+	usedCmd, err := f.GetCommand()
+	if err != nil {
+		fmtlog.Println(err)
+		os.Exit(-1)
+	}
+	// IF a KV Store is enable and no sub-command called in agrs
+	if kv != nil && usedCmd == traefikCmd {
```

Comment:
This feels a little weird, to have to check that no sub-command have been called.
Naively I would think that only the code specified in the sub-command would be called. Might need some refactoring in the future 👼